src/python/pants/backend/python/goals/run_python_source_integration_test.py

```diff
@@ -222,4 +222,34 @@ def test_local_dist() -> None:
             f"{tmpdir}/foo/main.py",
         ]
         result = run_pants(args)
-        assert result.stdout == "LOCAL DIST\n"
+        assert result.stdout == "LOCAL DIST\n", result.stderr
+
+
+def test_runs_in_venv() -> None:
+    # NB: We aren't just testing an implementation detail, users can and should expect their code to
```

Comment:
This is true of running a python source but not of running a PEX. Presumably for the latter you would want to run a PEX like a PEX will run in production. I think this just means the test function name is a little lacking though, you are testing running from a source here.

Reply:
This is inside the file `run_python_source_integration_test.py` though. So in that case, it should be clear.

Reply:
Ah, yes.